File: pset6/credit.py
from cs50 import get_int


def checksum(n):
    sum = 0
    while n != 0:
        d = 2*((n // 10) % 10)
        sum += (n % 10) + (d % 10) + (d // 10)
        n //= 100
    return sum % 10 == 0


while True:
    n = get_int("Number: ")
    if (n > 0):
        break

if n < 56E14 and n >= 51E14 and checksum(n):
    print("MASTERCARD\n")
elif ((n < 5E15 and n >= 4E15) or (n < 5E13 and n >= 4E13)) and checksum(n):
    print("VISA\n")
elif ((n < 38E13 and n >= 37E13) or (n < 35E13 and n >= 34E13)) and checksum(n):
    print("AMEX\n")
else:
    print("INVALID\n")

File: pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

bool checksum(long n);

int main(void)
{
    long n;
    do
    {
        n = get_long("Number: ");
    }
    while (n < 1);

    if (n < 56E14 && n >= 51E14 && checksum(n))
    {
        printf("MASTERCARD\n");
    }
    else if (((n < 5E15 && n >= 4E15) || (n < 5E13 && n >= 4E13)) && checksum(n))
    {
        printf("VISA\n");
    }
    else if (((n < 38E13 && n >= 37E13) || (n < 35E13 && n >= 34E13)) && checksum(n))
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}

bool checksum(long n)
{
    int sum = 0;
    int doubdig;
    for (long head = n / 10; head > 0; head = head / 100)
    {
        doubdig = 2 * (head % 10);
        sum = sum + doubdig % 10 + doubdig / 10;
    }
    for (long head = n; head > 0; head = head / 100)
    {
        sum = sum + head % 10;
    }
    return sum % 10 == 0;
}